### ISSUE TITLE
[JENKINS-19838] Ability to use separate polling queues per SCM type

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-06-a2#7034]
<p>Some SCMs are quite fast at pollings while others are very slow (Eg. Git is fast with low load on the server, while CVS is slow with high load on the server).</p>

<p>It would be nice to have separate polling queues per SCM type, so the fast Git pollings are not blocked by the slow ones.</p>

---
<details><summary><i>Originally reported by <a title="Sascha Vogt " href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=saschavogt">saschavogt</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-19838" target="_blank">Ability to use separate polling queues per SCM type</a></i></summary>
<i><ul>
<li><b>reported by</b>: saschavogt
<li><b>status</b>: Open
<li><b>priority</b>: Major
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 0
<li><b>watchers</b>: 2
<li><b>imported</b>: 2025-11-06
</ul></i>
</details>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=saschavogt">saschavogt</a>:</i>
<p>Proposed pull request: <a href="https://github.com/jenkinsci/jenkins/pull/963" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/963</a></p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kohsuke">kohsuke</a>:</i>
<p>Pull request had a lively conversation about how to go about this.</p>
